.github/workflows/pr.yml

```diff
@@ -0,0 +1,58 @@
+name: "PR Tests"
+permissions: read-all
+
+# Trigger for PR an merge to develop branch
+on:
+  push:
+    branches: develop
+  pull_request:
+  workflow_dispatch:
+
+env:
+  CTEST_OUTPUT_ON_FAILURE: 1
+  LAPACK_VERSION: 3.12.0
+  PARALLEL: -j 2
+
+jobs:
+  mkl:
+    runs-on: intel-ubuntu-latest
+    # One runner for each domain
+    strategy:
+      matrix:
+        domain: [blas, dft, lapack, rng]
+    name: MKL ${{ matrix.domain }} CPU
+    steps:
+    - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+    - name: Restore netlib from cache
+      id: cache-lapack
+      uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      with:
+        path: lapack/install
+        key: lapack-${{ env.LAPACK_VERSION }}
+    - name: Install netlib
+      if: steps.cache-lapack.outputs.cache-hit != 'true'
+      run: |
+        curl -sL https://github.com/Reference-LAPACK/lapack/archive/refs/tags/v${LAPACK_VERSION}.tar.gz | tar zx
+        SHARED_OPT="lapack-${LAPACK_VERSION} -DBUILD_SHARED_LIBS=on -DCBLAS=on -DLAPACKE=on -DCMAKE_INSTALL_PREFIX=${PWD}/lapack/install"
+        # 32 bit int
+        cmake ${SHARED_OPT} -B lapack/build32 
+        cmake --build lapack/build32 ${PARALLEL} --target install
+        # 64 bit int
+        cmake ${SHARED_OPT} -DBUILD_INDEX64=on -B lapack/build64
+        cmake --build lapack/build64 ${PARALLEL} --target install
+    - name: Install oneapi
+      uses: rscohn2/setup-oneapi@2ad0cf6b74bc2426bdcee825cf88f9db719dd727 # v0.1.0
```

Comment:
It looks like we'll need to update this workflow for every public release of mkl/icx. Will this uses line also need to be updated each time the version of mkl/icx changes? Is there a way to somehow signify "latest" that doesn't require regularly updating this workflow? Thanks!

Reply:
`rscohn2/setup-oneapi@v0`
would pick up the latest. I use hashes because I believe the openssf scanning tool flags anything that is not a hash.  After we move the repo to onemkl-project org, we could decide to allow versions instead of hashes. Pinning the CI to a specific compiler/MKL makes CI more stable/reproducible so I recommend leaving it as is.

Reply:
Should we use faceless account for such automation?

Reply:
I understand why we need it, but as far as I know using personal id could be a problem for security scans, but I'm not sure if it's a case here.

Reply:
I took a couple repos through the openssf badge process which includes openssf best practices. There were several requirements related to github actions, but I don't think there is anything about which org a GitHub Action comes from. The rules for private repos with access to intel resources are different, but this is a public repo. In any case, if there is a problem in the future, I will fix it.